kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/inband/InBandKeyManager.java

```diff
@@ -110,18 +117,57 @@ private CompletableFuture<KeyContext> makeKeyContext(@NonNull K kekId) {
                 }).toCompletableFuture();
     }
 
-    @NonNull
     @Override
+    @NonNull
     @SuppressWarnings("java:S2445")
-    public CompletionStage<Void> encrypt(@NonNull String topicName,
-                                         int partition,
-                                         @NonNull EncryptionScheme<K> encryptionScheme,
-                                         @NonNull List<? extends Record> records,
-                                         @NonNull Receiver receiver) {
-        if (records.isEmpty()) {
-            return CompletableFuture.completedFuture(null);
+    public CompletionStage<MemoryRecords> encrypt(@NonNull String topicName,
+                                                  int partition,
+                                                  @NonNull EncryptionScheme<K> encryptionScheme,
+                                                  @NonNull MemoryRecords records,
+                                                  @NonNull Function<Integer, ByteBufferOutputStream> bufferAllocator) {
+        if (records.firstBatch() == null) {
+            return CompletableFuture.completedFuture(records);
```

Comment:
What should we complete with here? null or the input MemoryRecords? Both work I think, but with null we would need a check in the Filter to not call set on the partitionProduceData if the result is null.

Reply:
Let's return the `records`.

Reply:
done, softened the javadoc to say it completes with MemoryRecords, rather than specifying that it is a new instance